src/main/java/com/github/charithe/kafka/KafkaJunitRule.java

```diff
@@ -84,8 +90,12 @@ public KafkaJunitRule(final int kafkaPort) {
 
     @Override
     protected void before() throws Throwable {
-        zookeeper = new TestingServer(true);
-        zookeeperPort = zookeeper.getPort();
+        if (zookeeperPort == -1) {
```

Comment:
Please change to `zookeeperPort == ALLOCATE_RANDOM_PORT` to be consistent with how Kafka random ports are allocated.